Modules/_threadmodule.c

```diff
@@ -42,24 +42,76 @@ get_thread_state(PyObject *module)
 
 // _ThreadHandle type
 
+// Handles transition from RUNNING to one of JOINED, DETACHED, or INVALID (post
+// fork).
+typedef enum {
+    THREAD_HANDLE_RUNNING,
+    THREAD_HANDLE_JOINED,
+    THREAD_HANDLE_DETACHED,
+    THREAD_HANDLE_INVALID,
+} ThreadHandleState;
```

Comment:
Explicitly assign values to these (non-0 to help differentiate from "uninitialized"?) so that people debugging crash dumps know for sure what is what by reading the code without having to understand what values the compiler auto-assigned to names. (I'm assuming C even specifies how auto-enumerating works -- 0 start and +1 each time IIRC -- but don't want anyone but a compiler maintainer to _need_ to know that)